src/Compilers/Shared/CoreClrShim.cs

```diff
@@ -15,5 +16,15 @@ internal static class AssemblyLoadContext
             internal static readonly Type Type = ReflectionUtilities.TryGetType(
                "System.Runtime.Loader.AssemblyLoadContext, System.Runtime.Loader, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
         }
+
+        internal static class AppContext
+        {
+            internal static readonly Type Type = ReflectionUtilities.TryGetType(
+                "System.AppContext, System.AppContext, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
```

Comment:
CoreCLR loader will allow this to bind to a later version, so as long as project.json only goes up from this, it will be fine. It makes sense to ask for the lowest version you actually need.